site/content/en/docs/tasks/dev/testing.md

```diff
@@ -0,0 +1,157 @@
+---
+title: "Running tests in Kueue"
+linkTitle: "Running tests in Kueue"
+weight: 25
+description: >
+  Running tests in Kueue
+---
+
+## Running tests in Kueue
+
+### Running unit tests
+To run all unit tests:
+```shell
+make test
+```
+
+To run unit tests for webhooks:
+```shell
+go test ./pkg/webhooks
+```
+To run tests that match `TestValidateClusterQueue` regular expression [ref](https://pkg.go.dev/cmd/go#hdr-Testing_flags):
+```shell
+go test ./pkg/webhooks -run TestValidateClusterQueue
+```
+
+#### Running unit tests with race detection
+
+```shell
+# install go stress
+go install golang.org/x/tools/cmd/stress@latest
+# recompile with race instrumentation
+go test ./pkg/scheduler/preemption/ -race -c
+# it loops and reports failures
+stress ./preemption.test -test.run TestPreemption
+```
+
+### Running integration tests
+
+For running subset of tests see [Running subset of tests](#running-subset-of-integration-or-e2e-tests).
+
+#### Increase logging verbosity
+You can change log level in the framework (for example, change -3 to -5 to increase verbosity):
+https://github.com/kubernetes-sigs/kueue/blob/f8015cb273f9115c34f9be32b35f7e1308c16459/test/integration/framework/framework.go#L72:
+```go
+var setupLogger = sync.OnceFunc(func() {
+	ctrl.SetLogger(util.NewTestingLogger(ginkgo.GinkgoWriter, -3))
+})
+```
+
+### Running e2e tests
+```shell
+# build an image
+export IMAGE_REGISTRY=<image registry to which you have push access>
+make image-local-push
+# in the output you can image path that looks like the following:
+# <image registry>/kueue:<tag>
+# or, you can get it by combining $IMAGE_REGISTRY:`git describe --tags --dirty --always`
+# pull that image:
+docker pull <image registry>/kueue:<tag>
+# run e2e test:
+make test-e2e
+# make test-tas-e2e
+# make test-multikueue-e2e
+```
```

Comment:
Instead of this it's much easier to use `make kind-image-build test-e2e`

Reply:
Or we may add a bit more details:

Step 1:
```sh
make kind-image-build
```
Step 2 against a new cluster:
```sh
IMAGE_TAG=us-central1-docker.pkg.dev/k8s-staging-images/kueue/kueue:<image version> CREATE_KIND_CLUSTER=true E2E_K8S_VERSIONS=1.32.0 make test-e2e
```
Or, step 2 against a pre-existing cluster:
```sh
kind load docker-image us-central1-docker.pkg.dev/k8s-staging-images/kueue/kueue:<image version> --name kind
IMAGE_TAG=us-central1-docker.pkg.dev/k8s-staging-images/kueue/kueue:<image version> CREATE_KIND_CLUSTER=true E2E_K8S_VERSIONS=1.32.0 make test-e2e
```